roms_tools/tests/test_utils.py

```diff
@@ -19,3 +29,80 @@ def test_coordinate_range_monotonicity(min_val, max_val, center, sc, N):
     )
     assert np.all(np.diff(faces) > 0), "faces is not strictly increasing"
     assert np.all(np.diff(centers) > 0), "centers is not strictly increasing"
+
+
+def test_has_dask() -> None:
+    """Verify that dask existence is correctly reported when found."""
+    with mock.patch("roms_tools.utils.find_spec", return_value=mock.MagicMock):
+        assert _has_dask()
+
+
+def test_has_dask_error_when_missing() -> None:
+    """Verify that dask existence is correctly reported when not found."""
+    with mock.patch("roms_tools.utils.find_spec", return_value=None):
+        assert not _has_dask()
+
+
+def test_has_gcfs() -> None:
+    """Verify that GCFS existence is correctly reported when found."""
+    with mock.patch("roms_tools.utils.find_spec", return_value=mock.MagicMock):
+        assert _has_gcsfs()
+
+
+def test_has_gcfs_error_when_missing() -> None:
+    """Verify that GCFS existence is correctly reported when not found."""
+    with mock.patch("roms_tools.utils.find_spec", return_value=None):
+        assert not _has_gcsfs()
+
+
+def test_load_data_dask_not_found() -> None:
+    """Verify that load data raises an exception when dask is requested and missing."""
+    with (
+        mock.patch("roms_tools.utils._has_dask", return_value=False),
+        pytest.raises(RuntimeError),
+    ):
+        _load_data("foo.zarr", {"a": "a"}, use_dask=True)
+
+
+def test_load_data_open_zarr_without_dask() -> None:
+    """Verify that load data raises an exception when zarr is requested without dask."""
+    with (
+        mock.patch("roms_tools.utils._has_dask", return_value=False),
+        pytest.raises(ValueError),
+    ):
+        # read_zarr should require use_dask to be True
+        _load_data("foo.zarr", {"a": ""}, use_dask=False, read_zarr=True)
+
+
+@pytest.mark.parametrize(
+    ("dataset_name", "expected_dim"),
+    [
+        ("surface_forcing", "time"),
+        ("bgc_surface_forcing", "time"),
+        # ("tidal_forcing", "eta_rho"),
```

Comment:
Is something wrong with the `tidal_forcing` fixture?

Reply:
i was testing 1 at a time and forgot to add that back. good :eye: !